### ISSUE TITLE
Priority ordering logic feels out of place

### ISSUE THREAD

**Issue by JoshOrndorff** [substrate-developer-hub/utxo-workshop#35]
There is a lot of custom logic for calculating priority of UTXO transactions in the tagged transaction queue runtime api https://github.com/substrate-developer-hub/utxo-workshop/blob/master/runtime/src/lib.rs#L339

This logic is specific to the utxo pallet, and it would feel more at home in the utxo pallet. One option would be to reduce the code in that runtime api to simply test whether the extrinsic is for UTXO pallet or not and either delegate to utxo pallet itself or to executive accordingly.

**Comment by nczhu** [comment]
Are you suggesting we make a utxo pallet? Or does one exist already?

Open to exploring what you are recommending both ways. I'm just a bit out of the loop! Thanks :)

**Comment by JoshOrndorff** [comment]
The UTXO pallet already exists in this file https://github.com/substrate-developer-hub/utxo-workshop/blob/master/runtime/src/utxo.rs

I'm suggesting (slash wondering) about this bunch of code https://github.com/substrate-developer-hub/utxo-workshop/blob/78135708fe787c1c13f5a9e16601f4b17bab53e4/runtime/src/lib.rs#L339-L399

It's logic is all about prioritizing UTXO transactions, so it would make sense that as much of it as possible be grouped with the UTXO pallet, rather than directly in the runtime's `lib.rs` file.

Maybe the part in the runtime's `lib.rs` could look like
```rust
if let Some(&utxo::Call::execute(ref transaction)) = IsSubType::<utxo::Module<Runtime>, Runtime>::is_sub_type(&tx.function) {
  // All the logic moves into this helper function in the pallet
  Utxo::validate_transaction(tx)
}
else {
  Executive::validate_transaction(tx)
}
```

**Comment by JoshOrndorff** [comment]
Also, that `if let` is complex and took me a while to wrap my head around. I found these useful in understanding it.
* https://substrate.dev/rustdocs/master/frame_support/dispatch/trait.IsSubType.html
* https://doc.rust-lang.org/rust-by-example/scope/borrow/ref.html

**Comment by nczhu** [comment]
Ah ok, there’s a miscommunication, I thought you meant this should be (already is) an official UTXO palette that’s published substrate 2.0. Yeah utxo.rs contains most of the runtime logic. 

And I agree that particular snippet can be refactored. We should be keeping lib.rs light/as boilderplate as possible, esp in this tutorial. 

Lemme think through where it should live though... I’m refactoring ALL of the code this week and trying to fix a critical issue, so a lot of this might change.
